prov/usnic/src/usdf_pep.c

```diff
@@ -337,6 +337,104 @@ usdf_pep_listen(struct fid_pep *fpep)
 	return 0;
 }
 
+/* Register as a callback triggered by the socket becoming writeable. Write as
+ * much data as can be written in a single write, and keep track of how much
+ * data is left. If the data is not fully written, it will finish getting
+ * written in another iteration of the progression.
+ */
+static int usdf_pep_reject_async(void *vreq)
+{
+	struct usdf_connreq *crp;
+	struct usdf_eq *eq;
+	int ret;
+
+	crp = vreq;
+	eq = crp->cr_pep->pep_eq;
+
+	do {
+		ret = write(crp->cr_sockfd, crp->cr_ptr, crp->cr_resid);
+	} while ((ret < 0) && (errno == EINTR || errno == EAGAIN));
```

Comment:
This looping on `EAGAIN` doesn't seem to agree with your comment above about "If the data is not fully written, it will finish getting written in another iteration of the progression."  Did you really mean to (potentially) block the progression loop?